src/network/networkd-mptcp.c

```diff
@@ -0,0 +1,390 @@
+                                         mp_tcp->section->filename, mp_tcp->section->line);
+
+        return 0;
+}
+
+int config_parse_mp_tcp_id(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+
+        _cleanup_(mp_tcp_free_or_set_invalidp) MPTCP *mp_tcp = NULL;
+        Network *network = data;
+        int r;
+
+        assert(filename);
+        assert(lvalue);
+        assert(rvalue);
+        assert(data);
+
+        r = mp_tcp_new_static(network, filename, section_line, &mp_tcp);
+        if (r < 0)
+                return r;
+
+        if (isempty(rvalue)) {
+                mp_tcp->id = (uint8_t) -1;
+
+                TAKE_PTR(mp_tcp);
+                return 0;
+        }
+
+        r = safe_atou8(rvalue, &mp_tcp->id);
+        if (r < 0) {
+                log_syntax(unit, LOG_WARNING, filename, line, r,
+                           "Failed to parse MPTCP '%s=', ignoring assignment: %s", lvalue, rvalue);
+                return 0;
+        }
+
+        mp_tcp->id_is_set = true;
+
+        TAKE_PTR(mp_tcp);
+        return 0;
+}
+
+int config_parse_mp_tcp_uint32(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+
+        _cleanup_(mp_tcp_free_or_set_invalidp) MPTCP *mp_tcp = NULL;
+        Network *network = data;
+        uint32_t k;
+        int r;
+
+        assert(filename);
+        assert(lvalue);
+        assert(rvalue);
+        assert(data);
+
+        if (isempty(rvalue)) {
+                if (streq(lvalue, "LimitSubflows"))
+                        network->manager->mp_tcp_subflows = (uint32_t) -1;
+                else if (streq(lvalue, "LimitAddress"))
+                        network->manager->mp_tcp_add_addr_accepted = (uint32_t) -1;
+                else
+                        assert_not_reached("Invalid lvalue");
+
+                TAKE_PTR(mp_tcp);
+                return 0;
+        }
+
+        r = safe_atou32(rvalue, &k);
+        if (r < 0) {
+                log_syntax(unit, LOG_WARNING, filename, line, r,
+                           "Failed to parse MPTCP '%s=', ignoring assignment: %s", lvalue, rvalue);
+                return 0;
+        }
+
+        if (streq(lvalue, "LimitSubflows")) {
+                network->manager->mp_tcp_subflows = k;
+        } else if (streq(lvalue, "LimitAddress"))
+                network->manager->mp_tcp_add_addr_accepted = k;
```

Comment:
hmm, wut? what's going on here? this is a per-network configuration option that actually controls a daemon global setting? Something is really off here...

Reply:
These are global settings not limited to a link. Hence so

Reply:
hmm, but then, why configure them in a .network file? what happens if i want to have multiple MTCP connections set up? Which one wins?

i.e. making a global setting locally configurable sounds wrong, no?